src/OrbitBase/File.cpp

```diff
@@ -195,23 +156,13 @@ ErrorMessageOr<size_t> ReadFully(const unique_fd& fd, void* buffer, size_t size)
 
 ErrorMessageOr<size_t> ReadFullyAtOffset(const unique_fd& fd, void* buffer, size_t size,
                                          int64_t offset) {
-  uint8_t* current_position = static_cast<uint8_t*>(buffer);
-  size_t bytes_read = 0;
-  while (bytes_read < size) {
-    int64_t pread_result = TEMP_FAILURE_RETRY(pread(fd.get(), current_position, size, offset));
-    if (pread_result == -1) {
-      return ErrorMessage{SafeStrerror(errno)};
-    }
-    if (pread_result == 0) {
-      break;
-    }
-    bytes_read += pread_result;
-    current_position += pread_result;
-    size -= pread_result;
-    offset += pread_result;
+  int64_t seek_result = lseek64(fd.get(), offset, SEEK_SET);
```

Comment:
I wondered if the seek needs to be wrapped in TEMP_FAILURE_RETRY. My understanding is that seek does not exhibit the EINTR behaviour as read/write. So this should be fine - but I'm a bit puzzeled why this is the case, thats why I mention it.

Reply:
Yeah, https://www.gnu.org/software/libc/manual/html_node/File-Position-Primitive.html
does not mention EINTR so we are fine.